src/mango/src/mango_cursor_text.erl

```diff
@@ -105,7 +107,8 @@ execute(Cursor, UserFun, UserAcc) ->
         query_args = QueryArgs,
         user_fun = UserFun,
         user_acc = UserAcc,
-        fields = Cursor#cursor.fields
+        fields = Cursor#cursor.fields,
+        execution_stats = mango_execution_stats:log_execution_start(Stats)
```

Comment:
I think a bit of redundancy here with the double `execution`. Maybe something like `mango_execution_stats:log_start`. Based on the module name we know its the execution stats.